tests/neo.UnitTests/Consensus/UT_Consensus.cs

```diff
@@ -100,35 +98,234 @@ public void ConsensusService_Primary_Sends_PrepareRequest_After_OnStart()
+            // Payload should also be forced, otherwise OnConsensus will not pass
+            commitPayload.PrevHash = UInt256.Zero;
+            Console.WriteLine($"\nNew Hash is {mockContext.Object.Block.GetHashData().ToScriptHash()}");
+
+            Console.WriteLine($"\nForcing block VerificationScript to {updatedContract.Script.ToScriptHash()}");
+            mockContext.Object.Block.Witness = new Witness { };
+            mockContext.Object.Block.Witness.VerificationScript = updatedContract.Script;
+            Console.WriteLine($"\nUpdating BlockBase Witness scripthash  is {mockContext.Object.Block.Witness.ScriptHash}");
+            Console.WriteLine($"\nNew Hash is {mockContext.Object.Block.GetHashData().ToScriptHash()}");
+
+            Console.WriteLine("\nCN6 simulation time");
+            // Here we used modified mockContext.Object.Block.GetHashData().ToScriptHash() for blockhash
+            actorConsensus.Tell(GetCommitPayloadModifiedAndSignedCopy(commitPayload, 5, kp_array[5], mockContext.Object.Block.GetHashData()));
+
+            Console.WriteLine("\nWait for subscriber Local.Node Relay");
+            var onBlockRelay = subscriber.ExpectMsg<LocalNode.Relay>();
+            Console.WriteLine("\nAsserting time was Block...");
+            var utBlock = (Block)onBlockRelay.Inventory;
+
+            Console.WriteLine($"\nAsserting block NextConsensus..{utBlock.NextConsensus}");
+            utBlock.NextConsensus.Should().Be(updatedContract.ScriptHash);
+
+            Console.WriteLine("\n==========================");
             // ============================================================================
             //                      finalize ConsensusService actor
             // ============================================================================
-
-            //Thread.Sleep(4000);
-            Sys.Stop(actorConsensus);
+            Console.WriteLine("Finalizing consensus service actor and returning states.");
             TimeProvider.ResetToDefault();
 
-            Assert.AreEqual(1, 1);
+            //Sys.Stop(actorConsensus);
+        }
+
+        public ConsensusPayload GetCommitPayloadModifiedAndSignedCopy(ConsensusPayload cpToCopy, ushort vI, KeyPair kp, byte[] blockHashToSign)
+        {
+            var cpCommitTemp = cpToCopy.ToArray().AsSerializable<ConsensusPayload>();
+            cpCommitTemp.ValidatorIndex = vI;
+            cpCommitTemp.ConsensusMessage = cpToCopy.ConsensusMessage.ToArray().AsSerializable<Commit>();
+            ((Commit)cpCommitTemp.ConsensusMessage).Signature = Crypto.Default.Sign(blockHashToSign, kp.PrivateKey, kp.PublicKey.EncodePoint(false).Skip(1).ToArray());
+            // Payload is not being signed by vI, since we are bypassing this check as directly talking to subscriber
+            return cpCommitTemp;
+        }
+
+        public ConsensusPayload GetPayloadAndModifyValidator(ConsensusPayload cpToCopy, ushort vI)
```

Comment:
The same as above, can we use more explicative param names?

Reply:
I current like these names, but fell free to suggest.

Reply:
The names are ok, the problem is that they are shortened.
Can you use payloadToCopy and validatorIndex?

Reply:
We could use consensusPayloadToCopy of validtorIndex, But I do not like index in long names.

I will add a summary on both methods.

Reply:
Done